### ISSUE TITLE
“Last updated by” Username is not being updated after adding a machine

### ISSUE THREAD

**Issue by RyanTG** [bpoore/pbm-react#279]
API problem or app?

To reproduce: add a machine to a location.

**Comment by bpoore** [comment]
I'm not certain how / when the location last updated timestamp / username is coming into play for the API, but we are only explicitly updating that field app-side when the user presses 'confirm location is up to date'. We could update that field app-side when a user adds / removes a machine from a location... though those changes will be wiped out when the app makes more API calls if the changes aren't being made via the API upon those actions for the location. I imagine those changes are being reflected and we're just not picking them up because the responses from the API don't include location level-information. 

The call to add a machine is giving us this back form the API: 
{"location_machine":{"id":76799,"created_at":"2019-11-17T15:46:13.770-08:00","updated_at":"2019-11-17T15:46:13.770-08:00","location_id":5335,"machine_id":2441,"condition":null,"condition_date":null,"ip":null,"user_id":447,"machine_score_xrefs_count":null,"last_updated_by_username":"bap","machine_conditions":[]}}

And deleting looks like this: 
{"msg":"Successfully deleted lmx #76799"}

Assuming what I've outlined will work for our needs, this will be a small change that we can get in before we look to do another release.

**Comment by bpoore** [comment]
https://github.com/bpoore/pbm-react/tree/update-last-updated-when-add-remove-machine addresses some of this. For removing a machine, this appears to be working well. 

My concern about the response coming from the API is real when looking at adding a machine though. Since adding a new machine requires the creation of a new lmx, we can't be just faking the new machine being added to the screen after the new machine is added. To get around this, we've been making a request after a machine is added for the details of the location after the addition of the machine. I added to the app state a fairly sloppy way to track the last edited field when a new machine has been added, but if you test this, see it working, leave the location screen and go back, you will see the changes are wiped out. We will need to have the API update the last_updated_by_username and date_last_updated fields when a new lmx is added for this to fully work in the app. 

While we're making these updates. Do we want the last updated by field to update if a comment is made about a machine or a score added to a machine at a location? These are additional actions that are being captured specific to a location, though those changes are so granular I'm not sure we would want to do that. 

I can merge my branch as it is now if we want the update to the location when a machine is added, though I think it would be potentially more confusing to a user to see that field update and then go back to what it was if they go back to the screen later.

**Comment by RyanTG** [comment]
Yes, I think the last updated but value should change when someone leaves a machine comment or high score. I see last updated as “this location was last touched on this date” and it’s an indicator that someone visited the location, which means the location list is likely up to date. 

I agree that it would be confusing to users to have that value revert when they leave and go back. 

So is the next step to update the API? If so, we should make an issue.

**Comment by bpoore** [comment]
Yes, we will need to make a small update to the API for this to work. We need the GET endpoint: `/api/v1/locations/:id.json` to return the updated date_last_updated and last_updated_by_username with fresh data after a machine is added to a location. 
<img width="677" alt="Screen Shot 2019-12-29 at 1 07 59 PM" src="https://user-images.githubusercontent.com/7720282/71562990-8f134d80-2a3d-11ea-9548-f76364d9fc1b.png">

Looks like the date is properly being updated, but you can see there is a stale value for last_updated_by_username. 

For deleting a machine, we are getting the right values back the requesting location data so we are good there. 

Looking at adding a comment or a score, the location values were properly updated when I added a score, but I didn't see an update when adding a comment to a machine: 
<img width="643" alt="Screen Shot 2019-12-29 at 1 20 27 PM" src="https://user-images.githubusercontent.com/7720282/71563308-224e8200-2a42-11ea-83d5-fb269e26e52c.png">

I will create an issue for the API
